contracts/SimpleRegistry.sol

```diff
@@ -98,7 +101,10 @@ contract SimpleRegistry is Owned, MetadataRegistry, OwnerRegistry, ReverseRegist
 		onlyOwnerOf(_name)
 		returns (bool success)
 	{
-		delete reverses[entries[_name].reverse];
+		if (keccak256(bytes(reverses[entries[_name].reverse])) == _name) {
+			emit ReverseRemoved(reverses[entries[_name].reverse], entries[_name].reverse);
+			delete reverses[entries[_name].reverse];
+		}
 		entries[_name].deleted = true;
```

Comment:
Can we put this other delete statement also inside the conditional check introduced in the `drop` function? `entries[_name].deleted = true;`
I think if the condition is not satisfied then nothing should be changed.

Reply:
I think it should be kept outside the if because we still want to mark the entry as deleted regardless of whether we remove any associated reversal.

Reply:
I also think this is fine — it's OK for `onlyOwnerOf(_name)` to modify a value of `entries[_name]` at any time; it's `reverses` which might be not the one under caller's control